src/usage_metrics/scripts/save_github_metrics.py

```diff
@@ -0,0 +1,135 @@
+"""This script pull github traffic metrics and saves them to a GC Bucket."""
+
+import json
+import logging
+import os
+import sys
+from dataclasses import dataclass
+from datetime import date
+
+import requests
+from google.cloud import storage
+from requests.exceptions import HTTPError
+
+logger = logging.getLogger()
+logging.basicConfig(level="INFO")
+
+
+@dataclass
+class Metric:
+    """Format metrics into folder names."""
+
+    name: str
+    folder: str
+
+
+TOKEN = os.getenv("API_TOKEN_GITHUB", "...")
+OWNER = "catalyst-cooperative"
+REPO = "pudl"
+BUCKET_NAME = "pudl-usage-metrics-archives.catalyst.coop"
+
+BIWEEKLY_METRICS = [
+    Metric("clones", "clones"),
+    Metric("popular/paths", "popular_paths"),
+    Metric("popular/referrers", "popular_referrers"),
+    Metric("views", "views"),
+]
+PERSISTENT_METRICS = [Metric("stargazers", "stargazers"), Metric("forks", "forks")]
+
+
+def get_biweekly_metrics(metric: str) -> str:
+    """Get json data for a biweekly github metric.
+
+    Args:
+        metric (str): The github metric name.
+
+    Returns:
+        json (str): The metric data as json text.
+    """
+    query_url = f"https://api.github.com/repos/{OWNER}/{REPO}/traffic/{metric}"
+    headers = {
+        "Authorization": f"token {TOKEN}",
+        "Accept": "application/vnd.github.v3+json",
+    }
+
+    response = make_github_request(query_url, headers)
+    return json.dumps(response.json())
+
+
+def get_persistent_metrics(metric) -> str:
+    """Get githubs persistent metrics: forks and stargazers.
+
+    Args:
+        metrics (str): the metric to retrieve (forks | stargazers)
+
+    Returns:
+        json (str): A json string of metrics.
+    """
+    query_url = f"https://api.github.com/repos/{OWNER}/{REPO}/{metric}"
+    headers = {
+        "Authorization": f"token {TOKEN}",
+        "Accept": "application/vnd.github.v3.star+json",
+    }
+
+    metrics = []
+    page = 1
+    while True:
```

Comment:
non-blocking: if somehow Github gives us an infinite stream of forks and stargazers (e.g. their pagination stops working for some reason??), we wouldn't want to keep requesting forever.

We could work around this by:
* limit how many pages to look at
* set a reasonable timeout on the job itself

In any case, the worst case scenario is the GHA runner eventually timing out after 6 hours of free compute. Hence, non-blocking.